tailscale/resource_tailnet_settings.go

```diff
@@ -0,0 +1,140 @@
+				Description: "Whether regional routing is enabled for the tailnet",
+				Optional:    true,
+			},
+			"posture_identity_collection_on": {
+				Type:        schema.TypeBool,
+				Description: "Whether identity collection is enabled for device posture integrations for the tailnet",
+				Optional:    true,
+			},
+		},
+	}
+}
+
+func resourceTailnetSettingsRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*Clients).V2
+
+	settings, err := client.TailnetSettings().Get(ctx)
+	if err != nil {
+		return diagnosticsError(err, "Failed to fetch tailnet settings")
+	}
+
+	settingsMap := map[string]any{
+		"devices_approval_on":                         settings.DevicesApprovalOn,
+		"devices_auto_updates_on":                     settings.DevicesAutoUpdatesOn,
+		"devices_key_duration_days":                   settings.DevicesKeyDurationDays,
+		"users_approval_on":                           settings.UsersApprovalOn,
+		"users_role_allowed_to_join_external_tailnet": string(settings.UsersRoleAllowedToJoinExternalTailnets),
+		"network_flow_logging_on":                     settings.NetworkFlowLoggingOn,
+		"regional_routing_on":                         settings.RegionalRoutingOn,
+		"posture_identity_collection_on":              settings.PostureIdentityCollectionOn,
+	}
+	return setProperties(d, settingsMap)
+}
+
+func resourceTailnetSettingsCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	if err := resourceTailnetSettingsDoUpdate(ctx, d, m); err != nil {
+		return err
+	}
+	d.SetId(createUUID())
+	return resourceTailnetSettingsRead(ctx, d, m)
+}
+
+func resourceTailnetSettingsUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	if err := resourceTailnetSettingsDoUpdate(ctx, d, m); err != nil {
+		return err
+	}
+	return resourceTailnetSettingsRead(ctx, d, m)
+}
+
+func resourceTailnetSettingsDoUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	var role *tsclient.RoleAllowedToJoinExternalTailnets
+	_role, ok := d.GetOk("users_role_allowed_to_join_external_tailnet")
+	if ok {
+		role = tsclient.PointerTo(tsclient.RoleAllowedToJoinExternalTailnets(_role.(string)))
+	}
+	settings := tsclient.UpdateTailnetSettingsRequest{
+		DevicesApprovalOn:                      optional[bool](d, "devices_approval_on"),
+		DevicesAutoUpdatesOn:                   optional[bool](d, "devices_auto_updates_on"),
+		DevicesKeyDurationDays:                 optional[int](d, "devices_key_duration_days"),
+		UsersApprovalOn:                        optional[bool](d, "users_approval_on"),
+		UsersRoleAllowedToJoinExternalTailnets: role,
+		NetworkFlowLoggingOn:                   optional[bool](d, "network_flow_logging_on"),
+		RegionalRoutingOn:                      optional[bool](d, "regional_routing_on"),
+		PostureIdentityCollectionOn:            optional[bool](d, "posture_identity_collection_on"),
```

Comment:
FWIW we could just update every field every time. Kinda just acting like a PUT. Fine either way.

Reply:
We can't actually. Some fields require special plans (e.g. regional routing) and some require special permissions (e.g. network flow logging).